### PR TITLE
feat: add personio channel ids

### LINKED ISSUE #78: Allow to pass in a channel ID for a specific job posting
Personio links to their own implementation like so:
https://satellytes.jobs.personio.de/job/325420?_pc=361856#apply

In order to track the origin of a link. here it's `361856` and it refers to a test channel. Everyone who submits the form after arriving with a specific channel id should submit that channel instead of the satellytes.com channel which is the default.

I'm not sure if we can validate that channel id but let's try if we can make this happen.

### PR DESCRIPTION
**Pull request by milljoniaer**
### What is included?
This PR adds personio channel id support for the career form.
The channel id is passed by adding a searchParam `_pc` to the url.

e.g. `?_pc=361856`
https://satellytescommain21751-featpersoniochannelidsinsearch.gtsb.io/career/staff-software-engineer/?_pc=361856
(this should be the test channel)

Fallback is `329206` (satellytes.com id)

ToDo:

- [x] check if application arrives in the correct channel

closes #78

### CODE REVIEW

**Comment by ghost** [comment]
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report
**satellytes.com-main**

:tada: Your build was successful! [See the Deploy preview here.](https://satellytescommain21751-featpersoniochannelidsinsearch.gtsb.io)

## Build Details

[View the build logs here.](https://gatsbyjs.com/dashboard/f215130d-15e3-4176-9252-e13166fe48de/sites/674be368-adf0-4ac5-8fae-f9a4493adc51/builds/fa38ee85-e4cc-4b1d-9eaa-d51601fbd9f4/details)

:clock1: Build time: **1m**

## Performance

### Lighthouse report

| Metric | Score |
| ----- | ---- |
| Performance |   :large_orange_diamond: 78   |
| Accessibility |   :green_heart: 97   |
| Best Practices |   :green_heart: 100   |
| SEO |   :large_orange_diamond: 79   |

:link: [View full report](https://build-fa38ee85-e4cc-4b1d-9eaa-d51601fbd9f4.gtsb.io/reports/lighthouse/index.html)

File: src/components/pages/career-details/career-details.tsx [hunk 1]
```diff
@@ -53,7 +58,7 @@ export const CareerDetails = ({
 
       <CareerForm
         company_id="41230"
-        recruiting_channel_id="329206"
+        recruiting_channel_id={personioChannel ?? '329206'}
```
Comment (georgiee) [inline src/components/pages/career-details/career-details.tsx]: please make the number a speaking constant like `DEFAULT_CHANNEL_ID` and reference it. It's much more readable than this.

**Review by georgiee — APPROVED** [review]
Hey thanks that looks great 🙌 I have one comment otherwise good!
I tested the submission with no id, wrong id and two different ids and everything works as expected. Let's merge once you have addressed that small constant thing.

A note on this:
In Personio you can create and maintain tracking links, it would be so nice to integrate the url schema you have developed for our website, but of course personio is not offering any way to customize that instead they happily create links that point to the automatically generated page we do not want to use .

File: src/components/pages/career-details/career-details.tsx (hunk 1 above)
Comment (milljoniaer) [reply src/components/pages/career-details/career-details.tsx]: looks better, I fixed it 👍

**Comment by milljoniaer** [comment]
Thank you for your recommendation, I changed it and will merge now.

**Pull request merged**